### PR TITLE
PR for merging Display Library

### PR DESCRIPTION
**Pull request by kitfre**
This is the first iteration of the display library for HaskellDO. This PR is meant to simply populate the repo with what I have done so far.

### CODE REVIEW

File: src/DisplayTypes.hs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+-- Displayable takes a type and wraps it in a Display
+-- Display will be used in HaskellDO to handle media rendering
+class Displayable a where
+  display :: a -> Display
+
+-- |
+-- DisplayType contains the allowed media to render 
+-- Any extensions to this library (i.e custom Display instances) should add a type here 
+-- E.g adding a display for latex would add the type DisplayLatex here
+data DisplayType
```
Comment (NickSeagull) [inline src/DisplayTypes.hs]: I think that this should not be that "open", we should have just:

- Text
- HTML
- SVG
- JS
- Image

We can add more types in the future, but we should not constraint users into adding their instance here.

File: src/Displayable.hs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+import Graphics.Rendering.Chart.Backend.Diagrams 
+import Data.Vinyl.TypeLevel (RecAll)
+import Text.Blaze.Html5 (toHtml)
+import Text.Blaze.Html.Renderer.Text (renderHtml)
+import Data.Text.Lazy (toChunks)
+import qualified Data.Vinyl.Functor as V
+import qualified Data.Text as T
+
+-- instance of Displayable of Record from Frames package
+instance (RecAll V.Identity r Show, ColumnHeaders r, AsVinyl r) => Displayable (Record r) where
```
Comment (NickSeagull) [inline src/Displayable.hs]: This should be in another file, so we can import Displayable from there, and users can do the same.

File: src/Displayable.hs [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+      html = (T.unpack . T.concat . toChunks . renderHtml) (toHtml $ show rec)
+
+instance Displayable (Renderable a) where 
+  display r = unsafePerformIO ret 
+    where 
+      ret = do 
+        fp <- storeChart r 
+        return $ Display DisplayChart fp 
+
+-- Chart helper functions for storing and retrieving FilePath
```
Comment (NickSeagull) [inline src/Displayable.hs]: Same thing with this

**Review by NickSeagull — CHANGES_REQUESTED** [review]

**Comment by kitfre** [comment]
I broke the Record and Renderable Displayable instances into two separate files, and added the DisplayTypes you asked for, let me know if there's anything else 👍

File: src/DisplayCharts.hs [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        return $ Display DisplaySVG fp 
+
+-- Chart helper functions for storing and retrieving FilePath
+tmpDir :: IO () 
+tmpDir = createDirectoryIfMissing False ".tmpImages"
+  
+storeChart :: Renderable a -> IO FilePath 
+storeChart c = do 
+  tmpDir 
+  let filename = ".haskll-do-chart.svg"
```
Comment (NickSeagull) [inline src/DisplayCharts.hs]: typo?

**Review by NickSeagull — CHANGES_REQUESTED** [review]
Love the changes! I think that there's a typo, could you fix it? 😄

**Review by NickSeagull — APPROVED** [review]

**Comment by NickSeagull** [comment]
Nice, thanks!

**Pull request merged**